include/hip/hcc_detail/device_functions.h

```diff
@@ -63,6 +63,22 @@ __device__ static inline unsigned int __clzll(long long int input) {
     return input == 0 ? 64 : input > 0 ? __builtin_clzl(input) : __builtin_clzl(~input);
 }
 
+__device__ static inline unsigned int __lastbit_u32_u64(unsigned long long int input) {
+  return input == 0 ? -1 : __builtin_ctzl(input);
+}
+
+__device__ static inline unsigned int __bitextract_u32(unsigned int src0, unsigned int src1, unsigned int src2) {
```

Comment:
We have a BFE instruction and an OCKL function that exposes it that should probably be used here.

Reply:
Well, unless the compiler matches it...

Reply:
Hi Brian, where can I find the instructions/functions are you referring to?

Reply:
Sorry about that.  I'm referring to the functions __ockl_bfe_{i32,u32}.